copy/single.go

```diff
@@ -593,6 +622,20 @@ type diffIDResult struct {
 	err    error
 }
 
+func (ic *imageCopier) getCompressionFromBlob(srcInfo types.BlobInfo) *compressiontypes.Algorithm {
```

Comment:
I think this can be a stand-alone function, not a method; that would make it more convenient to potentially unit-test.